Source/Particles/Gather/GetExternalFields.cpp

```diff
@@ -22,7 +22,7 @@ GetExternalEBField::GetExternalEBField (const WarpXParIter& a_pti, long a_offset
 
     AcceleratorLattice const & accelerator_lattice = warpx.get_accelerator_lattice(lev);
     if (accelerator_lattice.m_lattice_defined) {
-        d_lattice_element_finder = accelerator_lattice.GetFinderDeviceInstance(a_pti, static_cast<int>(a_offset));
+        *d_lattice_element_finder = accelerator_lattice.GetFinderDeviceInstance(a_pti, static_cast<int>(a_offset));
```

Comment:
Thanks for these changes! In completeness, does there need to be something to delete the instances created here? Could this be a `unique_ptr`(which should take care of the deletion)?

Reply:
Initially I tried with `unique_ptr`, but several compiler errors appeared wherever we were calling `GetExternalEBField` inside a lambda function.

The compiler was complaining that we were trying to use a deleted function: `error: use of deleted function`. Probably because the capture by value `[=]` attempts a copy, although I would not be 100% sure how to fix that. I also suspect that this may be why the other pointers in this same file are also raw pointers rather than `unique_ptr`.

Reply:
@dpgrote 

Do you see what could be causing the CI segfaults in the current implementation? Where should the allocation of the raw pointer be?

Reply:
It looks like the pointer can't work since the object is passed to the lambda. I can push a commit that I think should solve this. I would do `LatticeElementFinderDevice d_lattice_element_finder`, and add an internal flag to `LatticeElementFinderDevice` signaling whether it has been initialized.